components/config-provider/index.tsx

```diff
@@ -91,11 +96,13 @@ class ConfigProvider extends React.Component<ConfigProviderProps> {
     }
 
     return (
-      <ConfigContext.Provider value={config}>
-        <LocaleProvider locale={locale || legacyLocale} _ANT_MARK__={ANT_MARK}>
-          {children}
-        </LocaleProvider>
-      </ConfigContext.Provider>
+      <div dir={direction}>
```

Comment:
Should we add extra div to ConfigProvider?

Reply:
I couldn't find any proper element to add `dir` attribute to it, so i added this div.
Do you have any suggestion.

Reply:
And something i forgot to say is that i use `dir` attribute in less files to change style of component based on their direction.

Reply:
Try pass `dir` in React context to every component.

Reply:
![screenshots](https://user-images.githubusercontent.com/12451743/63220452-49946900-c19d-11e9-9d8e-3d79f91ec1ad.png)

or think about a form, if it wrapped with `<div dir="rtl" />` all components get `dir` value from context except those are not React component.
![screenshots2](https://user-images.githubusercontent.com/12451743/63220520-c1af5e80-c19e-11e9-843e-a74f760f9faf.png)

Reply:
another approach is like your suggestion just use `dir` in components and let the user to design their forms or group of elements with rtl direction.

Reply:
like [this pen ](https://codepen.io/saeedr/pen/YzKWLQL) i've just added `dir="rtl` ro `Form` element and everything is ok.

Reply:
Removed extra div on `ConfigProvider`.